pkg/services/accesscontrol/evaluator/evaluator.go

```diff
@@ -31,26 +33,50 @@ func Evaluate(ctx context.Context, ac accesscontrol.AccessControl, user *models.
 	return res, err
 }
 
+func ValidateScope(scope string) bool {
+	prefix := scope[:len(scope)-1]
+	return !strings.ContainsAny(prefix, "*?")
```

Comment:
Just came to think of this, I think we also want to make sure that if it ends with an asterisk, `prefix` is either empty or ends with `:` or `/`.

I also think, since you're calling this with

```go
if !ValidateScope(prefix) {
```

Are we removing the last character twice? Which I guess would break for `*`.